### PR TITLE
CMSIS-NN: 1xN conv support for MVE

### PR DESCRIPTION
**Pull request by felix-johnny**
1. MVE specific function for 1xN convolution.
2. Updated use of macro ARM_MATH_AUTOVECTORIZE
3. Updated history for new function that is added.

### CODE REVIEW

File: CMSIS/NN/Source/NNSupportFunctions/arm_nn_mat_mul_core_4x_s8.c [hunk 1]
```diff
@@ -57,40 +58,40 @@ arm_status arm_nn_mat_mul_core_4x_s8(const int32_t row_elements,
...
-           ,[row1] "+r"(ip_row_1)
-           ,[row2] "+r"(ip_row_2)
-           ,[row3] "+r"(ip_row_3)
-           ,[out0] "+Te"(acc_n0)
-           ,[out1] "+Te"(acc_n1)
-           ,[out2] "+Te"(acc_n2)
-           ,[out3] "+Te"(acc_n3)
-           :[cnt] "r"(row_elements)
-           :"q0","q1", "q2","q3","q4", "memory", "r14");
+#if defined(ARM_MATH_MVEI) && !defined(ARM_MATH_AUTOVECTORIZE)
```
Comment (christophe0606) [inline CMSIS/NN/Source/NNSupportFunctions/arm_nn_mat_mul_core_4x_s8.c]: I still think another flag is needed to select C or Asm (in case this Asm specific syntax is not supported on a given C compiler).
It is not the goal of ARM_MATH_AUTOVECTORIZE flag which can also be applied to C code.
ARM_MATH_AUTOVECTORIZE is used to select a version of the code which is working well with auto vectorizer (so no unrolling, not too many intrisinsics ...)

File: CMSIS/NN/Include/arm_nnfunctions.h [hunk 2]
```diff
@@ -78,12 +78,15 @@
...
    *
-   * Define macro ARM_MATH_DSP, If the silicon supports DSP instructions.
+   * Define macro ARM_MATH_DSP, If the silicon supports DSP instructions(DSP extension).
    *
    * - ARM_MATH_MVEI:
    *
    * Define macro ARM_MATH_MVEI, If the silicon supports M-Profile Vector Extension.
-   *
+
+   * - ARM_MATH_AUTOVECTORIZE
```
Comment (christophe0606) [inline CMSIS/NN/Include/arm_nnfunctions.h]: This flag is not related to ASM.
I need to document the meaning of this flag in arm_math.h
But today it is not yet applied very consistently in the DSP library.

File: CMSIS/NN/Source/NNSupportFunctions/arm_nn_mat_mul_core_4x_s8.c (hunk 1 above)
Comment (felix-johnny) [reply CMSIS/NN/Source/NNSupportFunctions/arm_nn_mat_mul_core_4x_s8.c]: @christophe0606 ok. i'll create a new one..
